src/dir.targets

```diff
@@ -10,6 +10,15 @@
   <Target Name="GetAssemblyVersion"
           Returns="$(AssemblyVersion)"/>
 
+  <!-- These gets overwritten when we restore the build tools and import the below targets -->
+  <Target Name="Build" Condition="!Exists('$(ToolsDir)Build.Common.targets')" >
+    <CallTarget Targets="_RestoreBuildTools" />
+  </Target>
+  <!-- Set a default clean target -->
+  <Target Name="Clean">
```

Comment:
A lot of the current quirkiness comes from being in a clean enlistment or is related to cleaning.  You wouldn't normally expect the RestoreBuildTools step to be tied to "Clean", but it currently is.  

Today's behavior: If you do not have the build tools available (in packages), and you attempt to "Clean" a project (msbuild /t:Clean), it will restore build tools, and then fail with an error message ("error : The build tools package was just restored and so we cannot continue the build of an individual project because targets from the build tools package were not able to be imported. Please retry
 the build the individual project again.").  The second attempt to clean will first run the "RestoreBuildTools" step (which it will skip), and then it will succeed in Cleaning.

The new behavior moves RestoreBuildTools from an InitialTarget, so it no longer runs as part of every msbuild command.  If you type "msbuild /t:Clean", it will just attempt a clean as we would expect.  "msbuild /t:Build" still exhibits the old behavior of requiring you to build twice.

@weshaggard, the reason I'm overriding the "Clean" target is so that you can run "msbuild /t:Clean" from an individual library and it will attempt to clean.  Without overriding the target, you get an annoying "error MSB4057: The target "Clean" does not exist in the project." message.

Of course, you could make the rational statement that you shouldn't be trying to "Clean" a clean build, but I think this scenario does occur where you have a clean repo and then try to clean out of habit.  It's a bit annoying to see an error message in that scenario.  Additionally, if you have previously built binaries, but packages are missing for some reason, you don't want to see an error when trying to clean.

The general guidance for msbuild is that "InitialTargets" are typically used for error checking. I think that making RestoreBuildTools an InitialTarget causes some unexpected behavior and we can have equivalent / sane functionality by making it a precursor to "Build".